File: sum-linked-list/SumLinkedList.java
public class SumLinkedList {
    static class Node {
        public Node next;
        public int value;
        public Node(int value) {
            this.value = value;
            this.next = null;
        }
    }

    public Node sumLinkedList(Node linkedListOne, Node linkedListTwo) {
        Node linkedListHead = new Node(0);
        Node current = linkedListHead;
        int carryOver = 0;
        while(linkedListOne != null || linkedListTwo != null || carryOver != 0) {
            int one = (linkedListOne != null) ? linkedListOne.value : 0;
            int two = (linkedListTwo != null) ? linkedListTwo.value : 0;
            int sum = one + two + carryOver;

            Node node = new Node(sum % 10);
            current.next = node;
            current = node;

            carryOver = sum / 10;

            linkedListOne = (linkedListOne != null) ? linkedListOne.next : null;
            linkedListTwo = (linkedListTwo != null) ? linkedListTwo.next : null;
        }
        return linkedListHead.next;
    }
}

File: sum-linked-list/SumLinkedList.swift
import Foundation

class SumLinkedList {
    class Node {
        var value: Int
        var next: Node?
        init(value: Int) {
            self.value = value
        }
    }

    func sumLinkedList(_ linkedListOne: Node, _ linkedListTwo: Node) -> Node? {
        var linkedListHead = Node(value: 0)
        var current = linkedListHead
        var carryOver = 0
        var headOne: Node? = linkedListOne
        var headTwo: Node? = linkedListTwo
        while headOne != nil || headTwo != nil || carryOver != 0 {
            var one = (headOne != nil) ? headOne!.value : 0
            var two = (headTwo != nil) ? headTwo!.value : 0
            var sum = one + two + carryOver;

            var newNode = Node(value: sum % 10)
            current.next = newNode
            current = newNode

            carryOver = sum / 10
            headOne = (headOne?.next != nil) ? headOne!.next : nil
            headTwo = (headTwo?.next != nil) ? headTwo!.next : nil
        }
        return linkedListHead.next!
    }
}